packages/next/pages/_document.tsx

```diff
@@ -605,7 +605,7 @@ export class Head extends Component<
     const filteredChildren: ReactNode[] = []
 
     React.Children.forEach(children, (child: any) => {
-      if (child.type === Script) {
+      if (typeof child.type === 'function' && child.type.name === 'Script') {
```

Comment:
I think the previous check of `child.type === Script` was correct as that ensures it is the `next/script` component and not a custom component named `Script` 

The issue seemed to be from the import for `next/script` in `_document` being local `../client/script` and the other being non-local causing different instances to be loaded. The tests passed because the monorepo version of Next.js is being used and webpack treated both instances the same.

We should be able to add a regression test for this by adding it to an isolated test (`test/e2e` or `test/production`)

We can also fix the different instances being loaded by updating this line:
https://github.com/vercel/next.js/blob/4a86a8ffb1ed9db223c05509844b39a2d8661cfc/packages/next/build/webpack-config.ts#L819

to: `/^(?:private-next-pages\/|next\/(?:dist\/pages\/|(?:app|document|link|image|constants|dynamic|script)$)|string-hash$)/`

Reply:
Thanks so much clarifying, I wasn't aware that different instances of the same import could have been the issue. 

Just fixed it and added a regression test 👍